lib/filterx/tests/test_expr_comparison.c

```diff
@@ -552,14 +559,13 @@ Test(expr_comparison, test_null_cases_type_aware_comparison)
 
   _assert_comparison(filterx_string_new("foobar", 6), filterx_null_new(), FCMPX_EQ | FCMPX_TYPE_AWARE, FALSE);
   _assert_comparison(filterx_string_new("foobar", 6), filterx_null_new(), FCMPX_LT | FCMPX_TYPE_AWARE, FALSE);
-  _assert_comparison(filterx_string_new("foobar", 6), filterx_null_new(), FCMPX_GT | FCMPX_TYPE_AWARE, TRUE);
+  _assert_comparison(filterx_string_new("foobar", 6), filterx_null_new(), FCMPX_GT | FCMPX_TYPE_AWARE, FALSE);
   _assert_comparison(filterx_string_new("foobar", 6), filterx_null_new(), FCMPX_NE | FCMPX_TYPE_AWARE, TRUE);
 }
 
 Test(expr_comparison, test_string_cases_type_aware_comparison)
 {
-  // TODO: fix float precision error in g_ascii_dtostr 3.14 -> "3.1400000000000000000000001"
-  // _assert_comparison(filterx_string_new("3.14", 4), filterx_double_new(3.14), FCMPX_EQ | FCMPX_TYPE_AWARE, TRUE);
+  _assert_comparison(filterx_string_new("3.14", 4), filterx_double_new(3.14), FCMPX_EQ | FCMPX_TYPE_AWARE, TRUE);
```

Comment:
How was this resolved?

Reply:
Now we compare the lhs and rhs type, and they are different. This means that we no longer compare them as strings, where this 000001 matters, but we compare them as doubles, where the DBL_EPSILON check works as intended.

Reply:
Thanks for the explanation.